api/src/main/java/net/kyori/adventure/util/RGBLike.java

```diff
@@ -54,4 +55,46 @@
    * @since 4.0.0
    */
   @IntRange(from = 0x0, to = 0xff) int blue();
+
+  /**
+   * Converts the color represented by this RGBLike to the HSV color space. Result values will be in the range [0, 1].
+   *
+   * @return an array of three elements containing the hue, saturation, and value (in that order), of the color represented by this RGBLike
+   * @since 4.6.0
+   */
+  default float@NonNull[] asHSV() {
```

Comment:
On second thought... do we want to add `HSVLike` instead of just returning a `float` array?
```java
interface HSVLike {
  float h();
  float s();
  float v();
}
```

Reply:
Would `HSVLike` be something `TextColor` would extend, or would it have a separate implementation? The problem with `TextColor` extending it would be we need to convert RGB -> HSV each time we call `h`, `s`, and `v` unless we cache/precompute the HSV values for every `TextColor`.

Reply:
No, standalone interface + class.